pkg/controller/job/job_controller.go

```diff
@@ -1021,7 +1006,7 @@ func (jm *Controller) trackJobStatusAndRemoveFinalizers(ctx context.Context, job
 				needsFlush = true
 				uncountedStatus.Succeeded = append(uncountedStatus.Succeeded, pod.UID)
 			}
-		} else if pod.Status.Phase == v1.PodFailed || considerTerminated {
+		} else if considerPodFailed || finishedCond != nil {
 			ix := getCompletionIndex(pod.Annotations)
```

Comment:
Add a comment: When the job is considered finished, every non-terminated pod is considered failed.

Reply:
done